config/config.example.yml

```diff
@@ -42,26 +49,134 @@ services:
+        # `/catalog/workspaces/user_xyz/dir1/dir2/subdir/file_abc`.
+        # The types of each segment of this target resource path  would be deduced from the `thredds_workspace` config below.
+        nginx_workspace:
+          - name: nginx
+            type: service
+          - name: workspaces
+            type: folder
+          - name: "{user}"
+            type: folder
+          - name: "**"
+            type: folder
+          - name: "{synched_file}"
+            type: file
+        thredds_workspace:
+          - name: catalog
+            type: service
+          - name: workspaces
+            type: folder
+          - name: "{user}"
+            type: folder
+          - name: "**"
+            type: folder
+          - name: subdir
+            type: folder
+          - name: "{synched_file}"
+            type: file
     # [Required] This section defines an array of permissions mapping between services
-    # In this example, for every Geoserver's `read` permission, a `read` and `browse` permissions will be created in
-    # Thredds. If a `browse` permission is created in Thredds a `read` permission will be added to `Geoserver`.
+    # Each item found in the `permissions_mapping` use the following format :
+    #     resource_key1 : permission(s) (-> or <- or <->) resource_key2 : permission(s)
+    # The arrows between the 2 resources indicate the direction of the synchronization, and which resources can be a
+    # `source` or `target` resource.
+    # In the case of `<->` arrow, the synchronization of permissions can be done in either direction. Also, it is
+    # important to note that in thise case, both mapped resources should matching `variable` names if any, and if one
+    # of the resource uses the `**` MULTI_TOKEN, the other resource should also include it in its path, to know how to
+    # match the segments.
+    # In the case of `->` or `<-` arrow, the synchronization is only done one way. In this case, the `source` resource
+    # path should include every `variable` names found in the `target` resource, but it can have more `variables` that
+    # just won't be used in the `target` path. Also, if the `target` resource uses the MULTI_TOKEN `**`, the `source`
+    # resource should have one too. The `source` can also use the `**` token even if the `target` doesn't include one.
+    permissions_mapping:
+      # A read permission for nginx_workspace creates/deletes both read and browse permissions in
+      # thredds_workspace. And a read or a browse permission on thredds_workspace creates/deletes a read
+      # permission on nginx_workspace.
+      - "nginx_workspace : read <-> thredds_workspace : [read,browse]"
+      # A write permission on nginx_workspace creates/deletes a write permission on thredds_workspace, and vice
+      # versa.
+      - "nginx_workspace : write <-> thredds_workspace : write"
+  weaver_outputs:
+    services:
+      wps:
```

Comment:
About this comment and the [other comment](https://github.com/Ouranosinc/cowbird/pull/21#discussion_r898323829) about the `weaver_wps_outputs` resource, I did change how the resources were organized temporarily without checking if it was valid, just so that it would work with our new constraint of using only actual Magpie service names here.

When I get service names from Magpie, I get those : `["access", "api", "geoserver", "geoserverwfs", "geoserverwms", "geoserverwps", "ncwms", "thredds", wfs", "wps"]`. 
`weaver` and `nginx` are not part of this list and it would result in a crash if I use those names, since it has to be an actual Magpie service. Does it make sense to use of those service names from the list here instead? Which one should I use?

Reply:
In this case they are only Mapgie services. Both are `api`. 
The weaver one manages endpoints behind weaver (processes, jobs, etc.), while the `weaver_wps_outputs` is the (not yet defined) `wps_outputs` service that will protect what is currently returned by nginx `wpsoutputs`. The config is planing ahead for https://crim-ca.atlassian.net/browse/DAC-469.